scripts/dep-supply-chain-check.py

```
@@ -0,0 +1,956 @@
+            if name and val.get("version"):
+                out[name] = str(val["version"])
+    deps = data.get("dependencies")
+    if isinstance(deps, dict):
+        for name, val in deps.items():
+            if isinstance(val, dict) and val.get("version"):
+                out.setdefault(name, str(val["version"]))
+    return out
+
+
+def npm_lock_root_deps(text):
+    """lock v2/v3 根包（packages[""]）的解析后直接依赖版本。"""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)